### ISSUE TITLE
TypeScript reporting a type checking error on a valid map initialization of an array of enum values

### ISSUE THREAD

**Issue by arobinson** [microsoft/TypeScript#31204]
**TypeScript Version:**  3.4.5

**Search Terms:**

Enum TypeError Map Array

**Code**

```ts
export enum AppType {
  HeaderDetail = 'HeaderDetail',
  HeaderMultiDetail = 'HeaderMultiDetail',
  AdvancedList = 'AdvancedList',
  Standard = 'Standard',
  Relationship = 'Relationship',
  Report = 'Report',
  Composite = 'Composite',
  ListOnly = 'ListOnly',
  ModuleSettings = 'ModuleSettings'
}

export enum AppStyle {
  Tree,
  TreeEntity,
  Standard,
  MiniApp,
  PivotTable
}

const appTypeStylesWithError: Map<AppType, Array<AppStyle>> = new Map([
[AppType.Standard, [AppStyle.Standard, AppStyle.MiniApp]],
[AppType.Relationship, [AppStyle.Standard, AppStyle.Tree, AppStyle.TreeEntity]],
[AppType.AdvancedList, [AppStyle.Standard, AppStyle.MiniApp]]
]);

const appTypeStylesWorking: Map<AppType, Array<AppStyle>> = new Map([
[AppType.Standard, new Array(AppStyle.Standard, AppStyle.MiniApp)],
[AppType.Relationship, [AppStyle.Standard, AppStyle.Tree, AppStyle.TreeEntity]],
[AppType.AdvancedList, [AppStyle.Standard, AppStyle.MiniApp]]
]);
```

**Expected behavior:**

The map with enumerations and a list of enumerations is valid.

**Actual behavior:**

TypeScript is reporting a typing error `Type 'AppStyle' is not assignable to type 'AppStyle.Standard | AppStyle.MiniApp'.`

The odd thing is that using `new Array` around the types (even just the first one) causes the error to go away.

**Playground Link:** 
https://www.typescriptlang.org/play/#src=export%20enum%20AppType%20%7B%0D%0A%20%20HeaderDetail%20%3D%20'HeaderDetail'%2C%0D%0A%20%20HeaderMultiDetail%20%3D%20'HeaderMultiDetail'%2C%0D%0A%20%20AdvancedList%20%3D%20'AdvancedList'%2C%0D%0A%20%20Standard%20%3D%20'Standard'%2C%0D%0A%20%20Relationship%20%3D%20'Relationship'%2C%0D%0A%20%20Report%20%3D%20'Report'%2C%0D%0A%20%20Composite%20%3D%20'Composite'%2C%0D%0A%20%20ListOnly%20%3D%20'ListOnly'%2C%0D%0A%20%20ModuleSettings%20%3D%20'ModuleSettings'%0D%0A%7D%0D%0A%0D%0Aexport%20enum%20AppStyle%20%7B%0D%0A%20%20Tree%2C%0D%0A%20%20TreeEntity%2C%0D%0A%20%20Standard%2C%0D%0A%20%20MiniApp%2C%0D%0A%20%20PivotTable%0D%0A%7D%0D%0A%0D%0Aconst%20appTypeStylesWithError%3A%20Map%3CAppType%2C%20Array%3CAppStyle%3E%3E%20%3D%20new%20Map(%5B%0D%0A%5BAppType.Standard%2C%20%5BAppStyle.Standard%2C%20AppStyle.MiniApp%5D%5D%2C%0D%0A%5BAppType.Relationship%2C%20%5BAppStyle.Standard%2C%20AppStyle.Tree%2C%20AppStyle.TreeEntity%5D%5D%2C%0D%0A%5BAppType.AdvancedList%2C%20%5BAppStyle.Standard%2C%20AppStyle.MiniApp%5D%5D%0D%0A%5D)%3B%0D%0A%0D%0Aconst%20appTypeStylesWorking%3A%20Map%3CAppType%2C%20Array%3CAppStyle%3E%3E%20%3D%20new%20Map(%5B%0D%0A%5BAppType.Standard%2C%20new%20Array(AppStyle.Standard%2C%20AppStyle.MiniApp)%5D%2C%0D%0A%5BAppType.Relationship%2C%20%5BAppStyle.Standard%2C%20AppStyle.Tree%2C%20AppStyle.TreeEntity%5D%5D%2C%0D%0A%5BAppType.AdvancedList%2C%20%5BAppStyle.Standard%2C%20AppStyle.MiniApp%5D%5D%0D%0A%5D)%3B%0D%0A

**Related Issues:**

**Comment by AlCalzone** [comment]
The map type seems to be inferred from the first argument, because you didn't provide one. This works:
```ts
const appTypeStylesWithError = new Map<AppType, Array<AppStyle>>([
  [AppType.Standard, [AppStyle.Standard, AppStyle.MiniApp]],
  [AppType.Relationship, [AppStyle.Standard, AppStyle.Tree, AppStyle.TreeEntity]],
  [AppType.AdvancedList, [AppStyle.Standard, AppStyle.MiniApp]]
]);
```

**Comment by weswigham** [comment]
It looks like we're inferring tuples for the arrays within the `Map` call, but those types are too specific, since we're just assigning to an `Array<AppStyle>`.

**Comment by ahejlsberg** [comment]
A simpler repro:

```ts
declare function foo<T>(...args: T[]): T[];
let b1: { x: boolean }[] = foo({ x: true }, { x: false });
let b2: boolean[][] = foo([true], [false]);  // Error
```

During type inference we form a union of all inferences made from object literals to a particular type parameter. However, we don't do the same for array literals. We should.
